docs/en/engines/database-engines/atomic.md

```diff
@@ -3,15 +3,60 @@ toc_priority: 32
 toc_title: Atomic
 ---
 
-
 # Atomic {#atomic}
 
-It supports non-blocking `DROP` and `RENAME TABLE` queries and atomic `EXCHANGE TABLES t1 AND t2` queries. `Atomic` database engine is used by default.
+It supports non-blocking [DROP](#drop-table) and [RENAME TABLE](#rename-table) queries and atomic [EXCHANGE TABLES t1 AND t2](#exchange-tables) queries. `Atomic` database engine is used by default.
 
 ## Creating a Database {#creating-a-database}
 
+``` sql
+    CREATE DATABASE test[ ENGINE = Atomic];
+```
+
+## Specifics and recommendations {#specifics-and-recommendations}
+
+### Table UUID {#table-uuid}
+
+All tables in database `Atomic` have persistent [UUID](../../sql-reference/data-types/uuid.md) and store data in directory `/clickhouse_path/store/xxx/xxxyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy/`, where `xxxyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy` is UUID of the table. 
+Usually, the UUID is generated automatically, but the user can also explicitly specify the UUID in the same way when creating the table (this is not recommended). To display the `SHOW CREATE` query with the UUID you can use setting [show_table_uuid_in_table_create_query_if_not_nil](../../operations/settings/settings.md#show_table_uuid_in_table_create_query_if_not_nil). For example:
+
 ```sql
-CREATE DATABASE test ENGINE = Atomic;
+CREATE TABLE name UUID '28f1c61c-2970-457a-bffe-454156ddcfef' (n UInt64) ENGINE = ...;
 ```
 
-[Original article](https://clickhouse.tech/docs/en/engines/database-engines/atomic/) <!--hide-->
+### DROP TABLE {#drop-table}
+
+On `DROP TABLE` no data is removed, database `Atomic` just marks table as dropped by moving metadata to `/clickhouse_path/metadata_dropped/` and notifies background thread. Running queries still may use dropped table. Table will be actually removed when it's not in use.
+
+### RENAME TABLE {#rename-table}
+
+`RENAME` queries are performed without changing UUID and moving table data. 
+
+!!! note "Note"
+    `RENAME` and `DROP` are execute without IStorage-level RWLocks. These queries do not wait for the completion of queries using the table and will be executed instantly. Data deletion occurs asynchronously in a background thread.
+
+### DELETE/DETACH {#delete-detach}
+
+`DELETE` and `DETACH` queries are executed asynchronously — waits for the running `SELECT` queries to finish but is invisible to the new queries.
```

Comment:
If synchronous mode is not enabled by `SYNC` modifier of `database_atomic_wait_for_drop_and_detach_synchronously` setting, then it does not wait for running `SELECT`s, `INSERT`s and other queries which are using the table